src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java

```diff
@@ -328,18 +353,21 @@ private void runOnDocker(DumbSlave s) throws Exception {
     private void runOnDocker(DumbSlave s, int sleepSeconds) throws Exception {
         j.jenkins.addNode(s);
         j.waitOnline(s);
-        FilePath dockerWS = s.getWorkspaceRoot();
+        FilePath dockerWs = s.getWorkspaceRoot();
         Launcher dockerLauncher = s.createLauncher(listener);
-        String script = String.format("echo hello world; sleep %s", sleepSeconds);
-        Controller c = new BourneShellScript(script).launch(new EnvVars(), dockerWS, dockerLauncher, listener);
-        while (c.exitStatus(dockerWS, dockerLauncher, listener) == null) {
-            Thread.sleep(100);
-        }
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        c.writeLog(dockerWS, baos);
-        assertEquals(0, c.exitStatus(dockerWS, dockerLauncher, listener).intValue());
-        assertTrue(baos.toString().contains("hello world"));
-        c.cleanup(dockerWS);
+
+        System.out.println("\n--smoke 10--");
+        smokeTest(dockerWs, dockerLauncher, sleepSeconds);
```

Comment:
Rather than running these all back-to-back in the same test, can we use [parameterized tests](https://github.com/junit-team/junit4/wiki/Parameterized-tests) so they run independently or at least [`ErrorCollector`](https://junit.org/junit4/javadoc/4.12/org/junit/rules/ErrorCollector.html) so that the test would run to completion and report all failures.

What I had in mind for parameterized tests would be to use something like an enum type to hold each platform (e.g. `NATIVE, ALPINE, CENTOS, UBUNTU, ...`), and then create a method `prepareAgentForPlatform(Platform p)` that just sets up an agent and adds it to Jenkins using the relevant Docker container (or `j.createOnlineSlave()` for `NATIVE`), returns some value containing the agent and things like the workspace and launcher (unless they can be computed using just the agent), and then the tests would call that method with the parameterized value and we can delete the `runOnFoo` tests and `runOnDocker`.